crypto/sleep.c

```diff
@@ -36,7 +37,11 @@ void OSSL_sleep(uint64_t millis)
     usleep(millis * 1000);
 #  endif
 # else
-    usleep(millis * 1000);
+    unsigned int s = (unsigned int)(millis / 1000);
```

Comment:
What happens on a system where `unsigned int` is 32 bits, but the value of `millis` is greater than 2^42 (approx)?

While it's clearly an edge case, someone might be pretty much expecting "sleep for ever" and get "return almost immediately"

Reply:
We did not check the overflow before with the usleep() call and do not for above usleep() calls either.

Reply:
Yes, but we do in the `_WIN32` case below - wouldn't this be a better pattern for all of the sleep implementations?

Reply:
Yeah that's right.

Reply:
I agree. Whether it needs to be done in this PR or whether it can be a separate one, that's another question. This one is already approved.